internal/boxcli/midcobra/telemetry.go

```diff
@@ -176,6 +181,9 @@ func (m *telemetryMiddleware) trackEvent(evt *event) {
 		_ = segmentClient.Close()
 	}()
 
+	// deliberately ignore error
+	_ = identifyUser(segmentClient, evt.DeviceID)
+
 	_ = segmentClient.Enqueue(segment.Track{ // Ignore errors, telemetry is best effort
```

Comment:
So, when the `userID` is available, I think we need to set it in `Track` too.

I think you need to refactor your code a bit:
- Add the `userID` field to event when we first create the event object (btw, that whole logic of creating the event object could now belong in it's own function)
- If the field is set, use it in both the call to `segment.Identify` and the call to `segment.Track`